p2p/conn/connection_bench_test.go

```diff
@@ -467,3 +467,102 @@ func BenchmarkMConnection_ScalingPayloadSizes_LowSendRate(b *testing.B) {
 		runBenchmarkTest(b, tt)
 	}
 }
+
+// BenchmarkMConnection_Multiple_ChannelID assesses the max bw/send rate
+// utilization of MConnection when configured with multiple channel IDs.
+func BenchmarkMConnection_Multiple_ChannelID(b *testing.B) {
+	// These tests create two connections with two channels each, one channel having higher priority.
+	// Traffic is sent from the client to the server, split between the channels with varying proportions in each test case.
+	// All tests should complete in 2 seconds (calculated as totalTraffic* msgSize / sendRate),
+	// demonstrating that channel count doesn't affect max bandwidth utilization.
+	totalTraffic := 100
+	msgSize := 1 * kibibyte
+	sendRate := 50 * kibibyte
+	recRate := 50 * kibibyte
+	chDescs := []*ChannelDescriptor{
+		{ID: 0x01, Priority: 1, SendQueueCapacity: 50},
+		{ID: 0x02, Priority: 2, SendQueueCapacity: 50},
+	}
+	type testCase struct {
+		name       string
+		trafficMap map[byte]int // channel ID -> number of messages to be sent
+	}
+	var tests []testCase
+	for i := 0.0; i < 1; i += 0.1 {
+		tests = append(tests, testCase{
+			name: fmt.Sprintf("2 channel IDs with traffic proportion %f %f",
+				i, 1-i),
+			trafficMap: map[byte]int{ // channel ID -> number of messages to be sent
+				0x01: int(i * float64(totalTraffic)),
+				0x02: totalTraffic - int(i*float64(totalTraffic)), // the rest of the traffic
+			},
+		})
+	}
+	for _, tt := range tests {
+		b.Run(tt.name, func(b *testing.B) {
+			for n := 0; n < b.N; n++ {
+				// set up two networked connections
+				// server, client := NetPipe() // can alternatively use this and comment out the line below
+				server, client := tcpNetPipe()
```

Comment:
I wonder if we could add latency accurately using a library such as [speedbump](https://github.com/kffl/speedbump/blob/master/lib/README.md)

Reply:
Thanks for the suggestion, will look into that https://github.com/celestiaorg/celestia-core/issues/1198